src/mesh/mesh_io.jl

```diff
@@ -196,6 +196,11 @@ function load_mesh_serial(mesh_file::AbstractString; n_cells_max, RealT)
     end
 
     mesh = CurvedMesh(size, mapping; RealT=RealT, unsaved_changes=false, mapping_as_string=mapping_as_string)
+  elseif mesh_type == "UnstructuredQuadMesh"
+    mesh_filename = h5open(mesh_file, "r") do file
+      return read(attributes(file)["mesh_filename"])
+    end
+    mesh = UnstructuredQuadMesh(mesh_filename; RealT=RealT, periodicity=false, unsaved_changes=false)
```

Comment:
Is `periodicity=false` always clear?

Reply:
For plotting purposes the type of boundary conditions are irrelevant. For example, it would be strange if the plotting routine needed to know if I used wall boundary conditions

Reply:
But this method is also used for restarts - as you did in an example here. The periodicity should be important there, isn't it?

Reply:
I just had a discussion with @andrewwinters5000 about how we could get rid of this `periodicity` argument by figuring out periodicity by the absence of non-periodic boundary names in the `.mesh` file, e.g., by doing something like
```julia
periodicity = issubset(boundary_names, map(Symbol, ["Periodic", "---"]))
```
in the `UnstructuredQuadMesh` ctor. However, since this is likely a larger change and requires more testing, @andrewwinters5000 suggested to just leave a `FIXME` comment here right now and he'll address this in the upcoming PR that works on the boundary name type instability.

If it is ok with you, I'd be ok in proceeding this way.

Reply:
Okay

Reply:
Coverage is OK. @andrewwinters5000 If you can add this `FIXME` comment and re-request my review, I"ll merge this PR immediately. Should we then also do a new Trixi release?

Reply:
Yeah, a new release would be good, I think.